ipaserver/install/installutils.py

```diff
@@ -259,6 +291,32 @@ def record_in_hosts(ip, host_name=None, conf_file=paths.HOSTS):
 
     return None
 
+
+def order_hosts_file(host_name):
+    """Read and overwrite /etc/hosts with FQDN at beginning, issue #8018"""
+    normalize_host = ''.join(host_name)
+    hostsfile = '/etc/hosts'
+    newfile = '/tmp/hosts'
+
+    with open(hostsfile, 'r') as hostsfp, open(newfile, 'w') as outfp:
+        for line in hostsfp:
+            if normalize_host in line:
+                restring = re.sub(r'%s( |$)' % normalize_host, '',
+                                  line.rstrip())
+                line = re.sub(r'(.*?) (.*)', r'\1 %s \2\n' % normalize_host,
+                              restring.rstrip())
+
+            outfp.write(line)
+
+    try:
+        shutil.copy(hostsfile, '%s.orig' % hostsfile)
+        shutil.move(newfile, hostsfile)
```

Comment:
After this you need to run `tasks.restore_context(hostsfile)`, to restore SELinux context. You also need to set permissions on the file. See, for example. `ipapython.ipautil.config_replace_variables()`.

Reply:
Use ``shutil.copystat()`` to copy ownership.

Reply:
done! copystat and also restore_context